### PR TITLE
[FIX] renderer: box clipRect includes cell borders

### PR DESCRIPTION
**Pull request by hokolomopo**
## Description:

There was a problem in the renderer, the cell content could be displayed
over its border. This was because the `clipRect` of the cell included
the cell's borders.

Odoo task ID : [3002154](https://www.odoo.com/web#id=3002154&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/plugins/ui_feature/renderer.ts [hunk 1]
```diff
@@ -733,4 +747,16 @@ export class RendererPlugin extends UIPlugin {
...
+
+  /**
+   * Computes the coordinates and size to draw the zone on the canvas, but exclude the borders of the cells.
+   */
+  private getVisibleRectWithoutBorders(zone: Zone, thinLineWidth: number): Rect {
+    const visibleRect = this.getters.getVisibleRect(zone);
+    return {
+      ...visibleRect,
+      height: visibleRect.height - thinLineWidth,
+      width: visibleRect.width - thinLineWidth,
```
Comment (LucasLefevre) [inline src/plugins/ui_feature/renderer.ts:759]: Why only `- thinLineWidth` and not `- 2*thinLineWidth`?
When I think about borders, I would think they are all around. It looks like something is wrong

Also, `thinLineWidth` does not make much sense here. Why does this method needs to know it's "thin" ? Something like `borderWidth` maybe ?

File: tests/plugins/renderer.test.ts [hunk 2]
```diff
@@ -297,12 +362,12 @@ describe("renderer", () => {
...
         {
           id: 1,
-          cols: { 0: { size: 5 }, 2: { size: 25 } },
+          cols: { 0: { size: 6 }, 2: { size: 26 } },
           colNumber: 3,
           cells: {
             A1: { content: "123456" },
             A2: { content: "=A1" },
-            C1: { content: "123456" },
+            C1: { content: "123458" },
```
Comment (anhe-odoo) [inline tests/plugins/renderer.test.ts:370]: Why did you changed these values ? (same for the setCellContent below)

**Comment by anhe-odoo** [comment]
Just two small comments:
1. Maybe use something related to "grid" instead of simply "border", to make a difference between the grid lines and the cell borders (as the future border PR will make us able to have other border width but it won't impact this fix)
2. I suppose that the commented code won't be kept, so don't forget to remove it :)

Else LGTM :)

File: tests/plugins/renderer.test.ts (hunk 2 above)
Comment (hokolomopo) [reply tests/plugins/renderer.test.ts:370]: To have an easier time at debugging and not have 2 cells with the same value.

**Comment by LucasLefevre** [comment]
Thanks for the work anyway

**Pull request closed without merging**